File: agenthub-web/src/types/agent.ts
export type Category = {
  id: number;
  name: string;
  icon?: string;
  sort?: number;
};

export type AgentCard = {
  id: number;
  userId: number;
  categoryId: number;
  categoryName?: string;
  name: string;
  description?: string;
  avatar?: string;
  type: 'PROMPT' | 'WORKFLOW' | string;
  visibility: string;
  starCount: number;
  forkCount: number;
  copyCount: number;
  viewCount: number;
  versionCount: number;
  score: number;
  parentAgentId?: number;
  authorName?: string;
  authorIcon?: string;
  isStar?: boolean;
  createTime?: string;
  updateTime?: string;
};

export type AgentVersion = {
  id: number;
  agentId: number;
  version: string;
  promptTemplate?: string;
  inputSchema?: string;
  workflowConfig?: string;
  modelSuggestion?: string;
  changelog?: string;
  createTime?: string;
};

export type AgentDetail = {
  agent: AgentCard;
  latestVersion?: AgentVersion;
  versions: AgentVersion[];
  viewTime?: string;
};

export type CopyResponse = {
  agentId: number;
  versionId: number;
  promptTemplate?: string;
  inputSchema?: string;
  workflowConfig?: string;
  modelSuggestion?: string;
};

export type FeedResponse = {
  list?: AgentCard[];
  minTime?: number;
  offset?: number;
};

export type AgentComment = {
  id: number;
  agentId: number;
  parentId?: number;
  userId: number;
  content: string;
  likes?: number;
  updated?: number;
  userName?: string;
  userIcon?: string;
  isLiked?: boolean;
  createTime?: string;
  replies?: AgentComment[];
};

File: src/main/java/com/agenthub/entity/Agent.java
package com.agenthub.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.io.Serializable;
import java.time.LocalDateTime;

@Data
@TableName("tb_agent")
public class Agent implements Serializable {
    private static final long serialVersionUID = 1L;

    @TableId(value = "id", type = IdType.AUTO)
    private Long id;
    private Long userId;
    private Long categoryId;
    private String name;
    private String description;
    private String avatar;
    private String type;
    private String visibility;
    private Integer status;
    private Integer starCount;
    private Integer forkCount;
    private Integer copyCount;
    private Integer viewCount;
    private Integer versionCount;
    private Integer score;
    private Long parentAgentId;
    private LocalDateTime createTime;
    private LocalDateTime updateTime;

    @TableField(exist = false)
    private String authorName;
    @TableField(exist = false)
    private String authorIcon;
    @TableField(exist = false)
    private String categoryName;
    @TableField(exist = false)
    private Boolean isStar;
}